src/averageAges.js

```diff
@@ -55,7 +78,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const femaleWomen = people.filter(person => person.sex === 'f');
+
+  const mothers = femaleWomen.filter(
+    women => people.some(person => person.mother === women.name));
+
+  const children = people.filter(
+    person => mothers.some(mother => person.mother === mother.name));
+
+  const sons = children.filter(child => child.sex === 'm');
```

Comment:
you can use one do it in one variable
like above
if with son check also sex in filter
then you can get average ages => use `map` for children and save child.born - people.find(here find mother of this child).born
Reduce it